source/postprocess/max_depth_field.cc

```diff
@@ -0,0 +1,131 @@
+
+      for (const auto &cell : this->get_dof_handler().active_cell_iterators())
+        if (cell->is_locally_owned())
+          {
+            fe_values.reinit (cell);
+
+            position_values = fe_values.get_quadrature_points();
+
+            for (unsigned int c = 0; c < this->n_compositional_fields(); ++c)
+              {
+                fe_values[this->introspection().extractors.compositional_fields[c]].get_function_values(this->get_solution(), compositional_values);
+
+                for (unsigned int q = 0; q < n_q_points; ++q)
+                  {
+                    const double depth = this->get_geometry_model().depth(position_values[q]);
+                    if (compositional_values[q] >= 0.5 && depth > local_max_depth[c])
+                      local_max_depth[c] = depth;
+                  }
+              }
+          }
+
+      // compute the max over all processors
+      std::vector<double> global_max_depth(this->n_compositional_fields());
+      Utilities::MPI::max (local_max_depth,
+                           this->get_mpi_communicator(),
+                           global_max_depth);
+
+      // finally produce something for the statistics file
+      for (unsigned int c = 0; c < this->n_compositional_fields(); ++c)
+        {
+          statistics.add_value("Max depth [km] for composition " + this->introspection().name_for_compositional_index(c),
+                               global_max_depth[c] / 1000.0);
+        }
+
+      // also make sure that the other columns filled by the this object
+      // all show up with sufficient accuracy and in scientific notation
+      for (unsigned int c = 0; c < this->n_compositional_fields(); ++c)
+        {
+          const std::string columns[] = {"Max depth [km] for composition " + this->introspection().name_for_compositional_index(c)};
+          for (unsigned int i = 0; i < sizeof(columns) / sizeof(columns[0]); ++i)
+            {
+              statistics.set_precision(columns[i], 8);
+              statistics.set_scientific(columns[i], true);
+            }
+        }
+
+      std::ostringstream output;
+      output.precision(4);
+      for (unsigned int c = 0; c < this->n_compositional_fields(); ++c)
+        {
+          output << global_max_depth[c] / 1000.;
+          if (c + 1 != this->n_compositional_fields())
+            output << " // ";
+        }
+      return std::pair<std::string, std::string>("Compositions max depth [km]:",
+                                                 output.str());
+
+    }
+  }
+}
+
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Postprocess
+  {
+    ASPECT_REGISTER_POSTPROCESSOR(MaxDepthField,
+                                  "maximum depth of field",
+                                  "A postprocessor that outputs the biggest depth "
+                                  "of the given field or of several given fields.")
```

Comment:
Would you mind how this is defined? Namely, the largest depth at which one can find a quadrature point where that field has a value of 0.5 or larger.

I would say "compositional field" rather than "given field".